pkg/steps/new_e2e.go

```diff
@@ -139,12 +145,26 @@ func (s newE2ETestStep) genPods(steps []api.TestStep) ([]coreapi.Pod, error) {
 		if owner := s.jobSpec.Owner(); owner != nil {
 			pod.OwnerReferences = append(pod.OwnerReferences, *owner)
 		}
+		if step.ArtifactDir != "" {
```

Comment:
Do we ever allow for an unset artifacts dir?

Reply:
"Unset" here means the flag wasn't present in the command line, which is a normal mode of operation. A more general question is: should we extract the information we need via artifacts, which I'll address below…